### ISSUE TITLE
Allow changing the capitalization in a username

### ISSUE THREAD

**Issue by nourkagha** [mastodon/mastodon#20487]
### Pitch

It should be possible to be able to adjust the capitalization (or case) of the letters in your username after it's created.

### Motivation

This is something that is offered by other services. It's currently not ideal that once you've made a username, you're stuck with it. Maybe you've decided that you want to go with a capital first letter rather than a lowercase one, or the other way around. This offers people choice and flexibility when it comes to how they would like to be represented.

**Comment by veale** [comment]
I think there is also an accessibility logic to this, as people may make a username without CamelCase and wish to change it later to be more accessible to those with screenreaders.

**Comment by miranda** [comment]
I would like to second this request, especially since usernames don't seem to be case-sensitive.  You should be able to change the capitalization pattern at will as long as they are the same letters.  Also, a lot of new users seem to be unclear on the difference between the username and display name when signing up.

**Comment by kudit** [comment]
Very much agree with this.

**Comment by thestinger** [comment]
@miranda

> I would like to second this request, especially since usernames don't seem to be case-sensitive. You should be able to change the capitalization pattern at will as long as they are the same letters. Also, a lot of new users seem to be unclear on the difference between the username and display name when signing up.

They're definitely already case insensitive. It's not mandated by the ActivityPub spec but that shouldn't matter since Mastodon already enforces uniqueness for this and the spec doesn't require immutability for this anyway. It's based on the URLs with the usernames only used to find those and it's Mastodon elevating them to be the main way to reference people and immutable.

**Comment by dmshaw** [comment]
Given that I found this suggestion specifically because I went searching to see if case changes were possible at all on Mastodon, I'm very much in favor of this idea, both for clarity at a glance and screen-reader readability reasons.

**Comment by yllekz** [comment]
Yes please. I made a casing mistake on my username a while back and really want to fix it. Account is far too established to delete/re-create either.

**Comment by thestinger** [comment]
The username will remain reserved if you delete the account so you would end up losing it rather than being able to fix it.

**Comment by fofwisdom** [comment]
When reattaching a deleted account with tootctl, there is a problem that the existing lowercase letters are reattached even though uppercase letters are inserted.
Even server administrators have no control over this issue.

**Comment by Lambdaphile** [comment]
+1

**Comment by intonarumori** [comment]
The signup form on mobile capitalises the username, so if you don't double check and/or don't know you cannot change it later, you have to run with it. I think this is a very bad default for new users and could be turned off with a very simple addition.

```
<input type="email" autocorrect="off" autocapitalize="none">
```

I've deleted my account in hopes I can register again with lowercase, and now lost the handle on that server.

**Comment by MarkSRobson** [comment]
This is essential for visually impaired users who rely on screen readers to find usernames, and, in my opinion, should have been an option from day one.

**Comment by gukj** [comment]
Out of interest, how does bad casing affect screen readers?

**Comment by dmshaw** [comment]
There is a good writeup at https://averment.medium.com/why-does-writing-your-hashtags-in-camel-case-make-them-more-accessible-and-what-are-the-benefits-9e3b8e13e920

The article is referring to hashtags, but the same concept applies to usernames.

**Comment by gukj** [comment]
@dmshaw @MarkSRobson Makes a lot of sense — thank you!

**Comment by t-offline** [comment]
Yes, please allow that.
I made a capitalization mistake in my username and want to fix it.

**Comment by Sweet-Light** [comment]
Due to this weird system, I have now lost my username. I created an account with capitalization I did not like, then discovered I could not change the capitalization, and deleted the account, hoping to recreate it with better capitalization. Now I find that my name is lost forever. Please allow to restore my name.

**Comment by LoganDark** [comment]
https://uwu.social/@logandark vs https://tech.lgbt/@LoganDark

**Comment by todb** [comment]
Pretty please.

**Comment by ivancarlosti** [comment]
+1 victim of autocapitalization
Please let us change it

**Comment by radoslew** [comment]
Please, allow us to change it! My OCD turns on because of it 😢

**Comment by sciencefictionspacecraft** [comment]
This seems sensible, doable, and wanted. But no progress in nearly four years...?

Please implement this feature, it's more important than you think....

**Comment by NeatNit** [comment]
> This seems sensible, doable, and wanted. But no progress in nearly four years...?
> 
> Please implement this feature, it's more important that you think....

This sometimes happens with FOSS projects. Even if it's easy (which we don't really know) and wanted by a lot of users, it might always stay just too far down the priority list for the maintainers.

This seems like a perfect case for a pull request. If you can code or willing to learn it, try to add this feature in yourself and create a PR for it.

FWIW, I am also annoyed that this feature is missing. But I understand that it's not really on the developers' radar.
